sql/core/src/test/scala/org/apache/spark/sql/UserDefinedTypeSuite.scala

```diff
@@ -134,6 +134,17 @@ class UserDefinedTypeSuite extends QueryTest with SharedSparkSession with Parque
     MyLabeledPoint(1.0, new TestUDT.MyDenseVector(Array(0.1, 1.0))),
     MyLabeledPoint(0.0, new TestUDT.MyDenseVector(Array(0.3, 3.0)))).toDF()
 
+
+  test("equal") {
```

Comment:
BTW, @Ngone51 . Although it seems to be considered as `Improvement` and should be `Improvement` due to the behavior change, it would be better if we have a test prefix `SPARK-32090: ` here in this PR. If you don't mind, please add it.

Reply:
Sure.